api/acl.go

```diff
@@ -443,18 +444,32 @@ func (a *ACLBindingRules) Get(bindingRuleID string, q *QueryOptions) (*ACLBindin
 }
 
 // ACLOIDC is used to query the ACL OIDC endpoints.
+//
+// Deprecated: ACLOIDC is deprecated, use ACLAuth instead.
 type ACLOIDC struct {
-	client *Client
+	ACLAuth
 }
 
 // ACLOIDC returns a new handle on the ACL auth-methods API client.
-func (c *Client) ACLOIDC() *ACLOIDC {
-	return &ACLOIDC{client: c}
+//
+// Deprecated: c.ACLOIDC() is deprecated, use c.ACLAuth() instead.
+func (c *Client) ACLOIDC() *ACLAuth {
```

Comment:
Although I think the methods still work, I'm pretty sure this changes the signature in a backwards incompatible way. Consider the case where someone has this code:

```go
var oidcAPI *ACLOIDC
oidcAPI = c.cACLOIDC

token , wm, err := oidcAPI.CompleteAuth(...)
```

It will no longer compile because of the type error and you'll get  `cannot use c.ACLOIDC() (value of type *ACLAuth) as *ACLOIDC value in assignment`. So we need to have `c.ACLOIDC` return `*ACLOIDC` still.

Reply:
oof, nice catch, thanks. I didn't consider it. Addressed in [36563a5](https://github.com/hashicorp/nomad/pull/16519/commits/36563a5c5587553d728028472ea4b2ec895e0bd8)